game/src/AI/States/SearchRadius.gd

```diff
@@ -0,0 +1,29 @@
+extends State
+"""
+State machine state that waits for the player to arrive to initiate a charging attack.
+Sets the target position to aim for to the straight line charge behavior.
+"""
+
+export var behavior: = NodePath()
+
+onready var _charge: StraightLineBehavior2D = get_node(behavior)
+
+var _target_shape: CollisionShape2D
+
+func physics_process(delta: float) -> void:
```

Comment:
You can remove physics process and charge in _on_Area2D_body_entered

Reply:
The issue is that if the player is already in the area2D by the time the search state rolls back around, the body_entered signal does not get emitted even when freshly connected by enter(). The physics_process function is there so it can trigger the state transition for that particular edge case.